assets/design.md

```diff
@@ -140,6 +151,31 @@ The `decorator` looks pretty similar to the `configurator` the only difference b
 
 #### Why do we need both Configurators and Decorators?
 
+There are two main reasons:
+
+- decoupleing configuration sources from manifest generation logic
+- working on a simplified representation of the model
+
+##### Decoupling configuration sources from manifest generator logic
+
+There are multiple configuration sources that contribute to the actual configuration that will be used for manfiest generation:
+
+- Annotations
+- Property configuration
+- Framework artifacts
+- Defaults
+
+Instead of coupling all those sources with the manifest generation logic, is preferable to limit their scope to the creation of the config object.
```

Comment:
```suggestion
Instead of coupling all those sources with the manifest generation logic, it is preferable to limit their scope to the creation of the config object.
```